Unable to build on Mac OS X

`fd_set` is no available on Mac

``` bash
➜  socks5_c git:(master) make
cc -Wall -O0 -DDEBUG=1 -g   -c -o local.o local.c
cc -Wall -O0 -DDEBUG=1 -g   -c -o buffer.o buffer.c
cc -Wall -O0 -DDEBUG=1 -g   -c -o encrypt.o encrypt.c
cc -Wall -O0 -DDEBUG=1 -g   -c -o socket_wrap.o socket_wrap.c
cc -Wall -O0 -DDEBUG=1 -g   -c -o socks.o socks.c
In file included from socks.c:1:
./socks.h:24:2: error: unknown type name 'fd_set'
        fd_set rfds, wfds;
        ^
./socks.h:27:2: error: unknown type name 'fd_set'
        fd_set _rfds, _wfds;
        ^
2 errors generated.
make: *** [socks.o] Error 1
```

--- a/socks.h
+++ b/socks.h
@@ -7,6 +7,7 @@
 #include "buffer.h"
 #include <stdint.h>
 #include <stdlib.h>
+#include <sys/select.h>
 
 #define AE_NONE 0
 #define AE_READABLE 1

